sources/core/objects/turfs/MetalWall.cpp

```diff
@@ -37,6 +41,59 @@ void MetalWall::AttackBy(IdPtr<Item> item)
     }
 }
 
+void MetalWall::AfterWorldCreation()
+{
+    UpdateState(Dir::ALL);
+}
+
+void MetalWall::Delete()
+{
+    NotifyNeighborhood();
+    MaterialObject::Delete();
+}
+
+void MetalWall::CheckNeighborhood(const Dir dir)
+{
+    if (dir == Dir::ALL)
+    {
+        for (const auto& x : WALL_PROCESSING_DIRS) CheckNeighborhood(x);
+        return;
+    }
+
+    if (IdPtr<MetalWall> wall = GetNeighbour(dir)->GetTurf())
+    {
+        SetBitByDirection(dir, true);
+    }
+    else
+    {
+        SetBitByDirection(dir, false);
+    }
+}
+
+void MetalWall::UpdateState(const Dir dir)
+{
+    CheckNeighborhood(dir);
+    if (dir == Dir::ALL)
+    {
+        NotifyNeighborhood();
```

Comment:
I don't think it is good that `UpdateState` can call `NotifyNeighborhood`. By its name `UpdateState` should work on the local state of the wall and not actively change states of neighbors.
As I can see, `UpdateState` with `dir == Dir::ALL` is called only in `AfterWorldCreation`, so it is possible to just call `NotifyNeighborhood` after `UpdateState` - that would be much clearer.